File: Two Sum.py
#https://leetcode.com/problems/two-sum/description/


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        #hashing
        dic={}
        for i in range(len(nums)):
            first=nums[i]
            second=target-first
            if second in dic.keys():
                return [i,dic[second]]
            dic[first]=i

File: Two Sum.cpp
//https://leetcode.com/problems/two-sum/description/


class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> search;
        vector<int> result;
        int valToFind;
        for(int i=0;i<nums.size();i++){
            valToFind=target-nums[i];
            auto find=search.find(valToFind);
            if(find!=search.end()){
                result.push_back(search[valToFind]);
                result.push_back(i);
                break;
            }
            search.insert({nums[i],i});
        }
        return result;
    }
};
